hpx/parallel/executors/persistent_auto_chunk_size.hpp

```diff
@@ -0,0 +1,126 @@
+//  Copyright (c) 2007-2015 Hartmut Kaiser
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+/// \file parallel/executors/persistent_auto_chunk_size.hpp
+
+#if !defined(HPX_PARALLEL_PERSISTENT_AUTO_CHUNK_SIZE_CONTROLLER_HPP)
+#define HPX_PARALLEL_PERSISTENT_AUTO_CHUNK_SIZE_HPP
+
+#include <hpx/config.hpp>
+#include <hpx/traits/is_executor_parameters.hpp>
+#include <hpx/runtime/serialization/serialize.hpp>
+#include <hpx/parallel/config/inline_namespace.hpp>
+#include <hpx/parallel/executors/executor_parameter_traits.hpp>
+#include <hpx/parallel/executors/executor_information_traits.hpp>
+#include <hpx/util/high_resolution_clock.hpp>
+#include <hpx/util/date_time_chrono.hpp>
+
+#include <cstddef>
+#include <algorithm>
+
+#include <boost/cstdint.hpp>
+
+namespace hpx { namespace parallel { HPX_INLINE_NAMESPACE(v3)
+{
+    ///////////////////////////////////////////////////////////////////////////
+    /// Loop iterations are divided into pieces and then assigned to threads.
+    /// The number of loop iterations combined is determined based on
+    /// measurements of how long the execution of 1% of the overall number of
+    /// iterations takes.
+    /// This executor parameters type makes sure that as many loop iterations
+    /// are combined as necessary to run for the amount of time specified.
+    ///
+    struct persistent_auto_chunk_size : executor_parameters_tag
+    {
+    public:
+        /// Construct an \a persistent_auto_chunk_size executor parameters object
+        ///
+        /// \note Default constructed \a persistent_auto_chunk_size executor parameter
+        ///       types will use 80 microseconds as the minimal time for which
+        ///       any of the scheduled chunks should run.
+        ///
+
```

Comment:
persistent_auto_chunk_size should have one input, which is the execution time of the chunk size, and by default it is zero. So min_time is not going to be used as an input and so by default I set it 80000 nanoseconds, the same as auto_chunk_size and it is going to be const here. Only execution time of the chunk size, which is chunk_size_time_ is changed.